src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -1544,14 +1544,23 @@ private static bool CanCreateNodeFromSelection(object parameter)
 
         /// <summary>
         /// Returns the selected nodes that are "input" nodes, and makes an 
-        /// exception for CodeBlockNodes as these are marked false so they 
-        /// do not expose a IsInput checkbox
+        /// exception for CodeBlockNodes and Filename nodes as these are marked 
+        /// false so they do not expose a IsInput checkbox
         /// </summary>
         /// <returns></returns>
         internal IEnumerable<NodeModel> GetInputNodesFromSelectionForPresets()
         {
             return DynamoSelection.Instance.Selection.OfType<NodeModel>()
-                                .Where(x => x.IsInputNode || x is CodeBlockNodeModel);
+                .Where(
+                    x => x.IsInputNode ||
```

Comment:
I'm wondering if this is a good situation for a nullable boolean. We could make `IsInput`, `bool?`. Then, for `CodeBlockNodeModel` and `FileName` nodes, we'd set `IsInput=null`. This would remove the need for any of this conditional code, allowing us to just say:
```c#
.Where(x=>x.IsInputNode.HasValue && x.IsInputNode)...
```